### PR TITLE
KAFKA-17857 Move AbstractResetIntegrationTest and subclasses to tools

### PR DESCRIPTION
**Pull request by brandboat**
related to https://issues.apache.org/jira/browse/KAFKA-17857,

as title, move AbstractResetIntegrationTest and subclasses to tools module

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -2500,6 +2502,7 @@ project(':tools') {
     testImplementation libs.apachedsLdifPartition
 
     testRuntimeOnly libs.junitPlatformLanucher
+    testRuntimeOnly libs.hamcrest
```
Comment (brandboat) [inline build.gradle:2519]: One wrinkle here is that org.apache.kafka.streams.integration.utils.IntegrationTestUtils relies on Hamcrest, so I add it to the tools module.

File: tools/src/test/java/org/apache/kafka/tools/AbstractResetIntegrationTest.java [hunk 2]
```diff
@@ -246,7 +244,7 @@ public void testReprocessingFromScratchAfterResetWithoutIntermediateUserTopic(fi
         final List<KeyValue<Long, Long>> resultRerun = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(resultConsumerConfig, OUTPUT_TOPIC, 10);
         streams.close();
 
-        assertThat(resultRerun, equalTo(result));
+        assertEquals(resultRerun, result);
```
Comment (m1a2st) [inline tools/src/test/java/org/apache/kafka/tools/AbstractResetIntegrationTest.java]: This `assertEquals` statement has the expected and actual values in the wrong order.

File: tools/src/test/java/org/apache/kafka/tools/AbstractResetIntegrationTest.java [hunk 3]
```diff
@@ -306,17 +304,17 @@ private void testReprocessingFromScratchAfterResetWithIntermediateUserTopic(fina
...
+        assertEquals(result, resultRerun);
+        assertEquals(result2, resultRerun2);
 
         if (!useRepartitioned) {
             final Properties props = TestUtils.consumerConfig(cluster.bootstrapServers(), appID + "-result-consumer", LongDeserializer.class, StringDeserializer.class, commonClientConfig);
             final List<KeyValue<Long, String>> resultIntermediate = IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(props, INTERMEDIATE_USER_TOPIC, 21);
 
             for (int i = 0; i < 10; i++) {
-                assertThat(resultIntermediate.get(i), equalTo(resultIntermediate.get(i + 11)));
+                assertEquals(resultIntermediate.get(i), resultIntermediate.get(i + 11));
```
Comment (m1a2st) [inline tools/src/test/java/org/apache/kafka/tools/AbstractResetIntegrationTest.java]: ditto This `assertEquals` statement has the expected and actual values in the wrong order.

**Review by m1a2st — COMMENTED** [review]
Hello @brandboat, Thanks for this PR, left some comments, PTAL

File: tools/src/test/java/org/apache/kafka/tools/AbstractResetIntegrationTest.java (hunk 3 above)
Comment (brandboat) [reply tools/src/test/java/org/apache/kafka/tools/AbstractResetIntegrationTest.java]: Thanks, I missed those two assertions.

**Comment by chia7712** [comment]
I have left some comment in jira.

**Comment by chia7712** [comment]
@brandboat could you please rebase code?

**Pull request merged**